content/docs/intro/concepts/programming-model.md

```diff
@@ -9,4529 +9,104 @@ menu:
-    Rucket:  role.ARN(),
-    Runtime: "python3.7",
-    Handler: "hello.handler",
-    Code:    fileArchive,
-})
-```
-
-{{% /choosable %}}
-{{% choosable language csharp %}}
-
-```csharp
-var fn = new Aws.Lambda.Function("fn", new Aws.Lambda.FunctionArgs
-{
-    Role = role.arn,
-    Runtime = "python3.7",
-    Handler = "hello.handler",
-    Code = fileArchive,
-});
-```
-
-{{% /choosable %}}
-
-{{< /chooser >}}
-
-### Runtime Functions
-
-The Pulumi SDK library also offers a number of helper functions.
-
-* [Current Project or Stack](#current-project-stack): get information about the current deployment
-* [Logging](#logging): for diagnostics, warnings, or errors pertaining to the deployment
-* [Serializing Lambdas](#runtime): for turning JavaScript callbacks into data which can be used as application code
-
-#### Getting the Current Project or Stack {#current-project-stack}
-
-The {{< pulumi-getproject >}} and {{< pulumi-getstack >}} functions give you the currently deploying project and stack, respectively. This can be useful for naming or tagging resources.
-
-{{< chooser language "javascript,typescript,python,go,csharp" >}}
-
-{{% choosable language javascript %}}
-
-```javascript
-let project = pulumi.getProject();
-let stack = pulumi.getStack();
-```
-
-{{% /choosable %}}
-{{% choosable language typescript %}}
-
-```typescript
-let project = pulumi.getProject();
-let stack = pulumi.getStack();
-```
-
-{{% /choosable %}}
-{{% choosable language python %}}
-
-```python
-project = pulumi.get_project()
-stack = pulumi.get_stack()
-```
+Although you use general-purpose languages, Pulumi is still a declarative infrastructure as code tool. After writing a program, you run the Pulumi CLI command `pulumi up`, which executes the program and determines the desired infrastructure state for all resources declared. The CLI will show you a preview of changes to be made, including all new resources to be created and existing resources to update or destroy. After confirming, Pulumi will carry out the changes.
 
-{{% /choosable %}}
-{{% choosable language go %}}
+**Stack**. [Stacks]({{< relref "/docs/intro/concepts/stack">}}) are an instantiation of a program and contain a collection of cloud resources. A project can have multiple stacks. Stacks are similar to environments. For example, a project might have a dev stack,  a test stack and a production stack.
 
-```go
-project := ctx.Project()
-stack := ctx.Stack()
-```
+**Resource**. A [resource]({{< relref "/docs/intro/concepts/resources">}}) represents a type of infrastructure, such as a compute instance. Resources can be aggregated together as component resources
 
-{{% /choosable %}}
-{{% choosable language csharp %}}
+**Resource Provider**. A [resource provider]({{< relref "/docs/intro/concepts/resource-providers">}}) A resource provider handles  communications  with a cloud service to create, read, update, and delete the resources you define in your Pulumi programs.
```

Comment:
To align with my suggestion on the diagram above, I think we should simplify this list and remove definitions below Resource for the purposes of this topic. Items like inputs and outputs can be referenced in other definitions with links to the various sub-topics for more information.